src/components/auth/AuthTorus.js

```diff
@@ -181,6 +176,55 @@ const AuthTorus = ({ screenProps, navigation, styles, store }) => {
   const facebookButtonHandler = useMemo(() => (asGuest ? signupFacebook : goToW3Site), [asGuest, signupFacebook])
   const facebookButtonTextStyle = useMemo(() => (asGuest ? undefined : styles.textBlack), [asGuest])
 
+  const auth0ButtonHandler = useMemo(() => (asGuest ? () => setPasswordless(true) : goToW3Site), [
+    asGuest,
+    signupAuth0,
+    setPasswordless,
+  ])
+
+  const signupAuth0Email = () => signupAuth0('email')
+  const signupAuth0Mobile = () => signupAuth0('mobile')
+
+  const ShowPasswordless = useMemo(
```

Comment:
i think better to move it outside, to the separate component because
a) it looks too large
b) it doesn't looks like some block / list we want to memoize
b) the component and it's callback are related to another service and another feature (auth0). When one component wants to use some other features - hooks or composition is a better way in React

Reply:
maybe, but again, its not a reusable component, it has only one single usage.
moving it outside would require passing 3 different callbacks as variables making it less readable.
sometimes when code is all in same place it easier to navigate and understand whats going on. when debuging need to pass through all the functions you lose track.
the zoomsdk is good example. its very hard for me to understand how exactly it works:)
so yeah its might not be tidy and neat but its easier to read

Reply:
okay, agreed
let's keep it here by now.
probably we would re-view it again when we'll have time available for that